### ISSUE TITLE
TimeoutError: waiting for selector "[aria-label="Listen only"]"

### ISSUE THREAD

**Issue by dimitrisal** [jibon57/bbb-recorder#10]
Hello i am trying to use the livejoin function, it load some page logs and then i get this error 

{ TimeoutError: waiting for selector "[aria-label="Listen only"]" failed: timeou                                                                                                                                                             t 30000ms exceeded
    at new WaitTask (/home/thomas/bbb-recorder/node_modules/puppeteer/lib/DOMWor                                                                                                                                                             ld.js:549:28)
    at DOMWorld._waitForSelectorOrXPath (/home/thomas/bbb-recorder/node_modules/                                                                                                                                                             puppeteer/lib/DOMWorld.js:478:22)
    at DOMWorld.waitForSelector (/home/thomas/bbb-recorder/node_modules/puppetee                                                                                                                                                             r/lib/DOMWorld.js:432:17)
    at Frame.waitForSelector (/home/thomas/bbb-recorder/node_modules/puppeteer/l                                                                                                                                                             ib/FrameManager.js:627:47)
    at Frame.<anonymous> (/home/thomas/bbb-recorder/node_modules/puppeteer/lib/h                                                                                                                                                             elper.js:112:23)
    at Page.waitForSelector (/home/thomas/bbb-recorder/node_modules/puppeteer/li                                                                                                                                                             b/Page.js:1122:29)
    at main (/home/thomas/bbb-recorder/liveJoin.js:66:20)
    at process._tickCallback (internal/process/next_tick.js:68:7) name: 'Timeout                                                                                                                                                             Error' }

My bbb server is HTML5 and using the greenlight 2.5.

Any ideas?

**Comment by jibon57** [comment]
Make sure that the server you are using for this purpose has set English language default. That error indicate that selector wasn't visible. May be loading take much time.

**Comment by dimitrisal** [comment]
The server os set in English but the HTML5 version( i dont know about the old one) is translating the components based on the browser / system locale . I did translate them but they didn't work , made few changes and final got a message REC_START

I before I see that message the following line had always problem , probably because the user still didn't connect in audio.
Line 75      await page.waitForSelector('button[aria-label="Leave audio"]');
No mater what I did it never worked I commented out.

I translated the line 86             await page.waitForSelector('[class^=modal] > [class^=content] > button[description="Logs you out of the meeting"]', {

and it did produce a recording but it didn't have any sound.

I translated the following lines , change from arial to label , add the button in click but no matter what i did the user still didn't connect with listen only
 await page.waitForSelector('[aria-label="Listen only"]');
 await page.click('[aria-label="Listen only"]', {waitUntil: 'domcontentloaded'});

**Comment by jibon57** [comment]
Working fine from me. Check using browser inspector to find if `aria-label="Listen only"` exists or not. Otherwise it won't auto click to go for listen only mood. If that was translated in different language then it won't show up. Because google chrome will visit that page & change language based on your server (`from where you are running the script not where BBB recording was`).

**Comment by awe007** [comment]
Good afternoon. I have exactly the same problem.

```
PAGE LOG: FFMPEG_SERVER
{ TimeoutError: waiting for selector "[aria-label="Listen only"]" failed: timeout 30000ms exceeded
    at new WaitTask (/root/bbb-recorder/node_modules/puppeteer/lib/DOMWorld.js:549:28)
    at DOMWorld._waitForSelectorOrXPath (/root/bbb-recorder/node_modules/puppeteer/lib/DOMWorld.js:478:22)
    at DOMWorld.waitForSelector (/root/bbb-recorder/node_modules/puppeteer/lib/DOMWorld.js:432:17)
    at Frame.waitForSelector (/root/bbb-recorder/node_modules/puppeteer/lib/FrameManager.js:627:47)
    at Frame.<anonymous> (/root/bbb-recorder/node_modules/puppeteer/lib/helper.js:112:23)
    at Page.waitForSelector (/root/bbb-recorder/node_modules/puppeteer/lib/Page.js:1122:29)
    at main (/root/bbb-recorder/liveRTMP.js:72:20)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7) name: 'TimeoutError' }
```

Did you manage to find a solution? If so, can you describe in detail what needs to be changed in order for the live broadcast to work?
Thank you in advance!

**Comment by awe007** [comment]
Please. help me :)

**Comment by dimitrisal** [comment]
Hello i didnt find any solution even though i changed the aria-label="Listen only" into the translated one still it didnt work so I didnt spent any more time in this since the main feature that i want is the converter

**Comment by jibon57** [comment]
`failed: timeout 30000ms`
That mean the script wait 30 seconds to get the page ready but failed. This mean the webpage wasn't open. This mostly happen for slow internet connection from where you are trying. You can adjust the timeout to your own needs like this:

https://github.com/jibon57/bbb-recorder/blob/235b85a9b065418358b40262195955efdf0cdd5e/liveRTMP.js#L72

```
await page.waitForSelector('[aria-label="Listen only"]', {
	timeout: 0 // for unlimited duration or 60000 for 60 seconds
});
```

**Comment by awe007** [comment]
I apologize in advance that there are so many questions, but we really need to launch a live broadcast. Very.
So I will be glad of any help.

Slow Internet is excluded. I have everything running locally. BigBlueButton is located on one server, and I am trying to get a live stream there.

In one window of the terminal, I run the command
`node ffmpegServer.js`

In another terminal window, I run the command
`node liveRTMP.js "https://tochkavstrechi.online/bigbluebutton/api/join?meetingId=adm-96w-m2a"`

Am I spelling MeetingID incorrectly ? Where can I view it on BigBlueButton?
And how can I manually check whether chrome gets [aria-label="Listen only"] ? After all, I am in the console, without a graphical interface.

And please tell me, if the command is successful, where and what link can I watch the live broadcast?

**Comment by mhd-shf** [comment]
I have the same problem,
The problem is not related to translation aria-labels or finding them, I captured screen step by step via below code and find that the user can't connect to audio:
`await page.screenshot({
            fullPage: true,
            path:`screenshot.png`
        });`

![image](https://user-images.githubusercontent.com/12235182/86467041-ea562d80-bd49-11ea-9b07-97591293774e.png)

![image](https://user-images.githubusercontent.com/12235182/86466757-774cb700-bd49-11ea-999a-6bc70710edba.png)

![image](https://user-images.githubusercontent.com/12235182/86467326-749e9180-bd4a-11ea-8806-6596d2230388.png)

As you can see, eventually the audio not connected and this is the root of problem. 
I have ubuntu 16.04 and the strange thing is that I tested this code on 3 other servers and works perfectly but I don't know why here doesn't work and the audio doesn't connect.

any idea?
